### PR TITLE
Inspector should not hold callback from garbage collection.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 1]
```diff
@@ -797,8 +796,9 @@ mixin WidgetInspectorService {
   void registerServiceExtension({
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:796]: make this method protected as it doesn't make sense to call outside of this class or subclasses.

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 2]
```diff
@@ -797,8 +796,9 @@ mixin WidgetInspectorService {
...
     required String name,
     required ServiceExtensionCallback callback,
+    required RegisterServiceExtensionCallback registration,
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:799]: the name `registration` is a little confusing to me as the tense seems odd. I don't have a great name idea but, I would err on the side of making it clear but a bit verbose. For example:
registerServiceExtensionCallback 
alternatives:
* `onRegisterServiceExtension`
* `serviceExtensionCallback`
* `registerExtension`
* `extensionRegistrar`

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 3]
```diff
@@ -809,8 +809,9 @@ mixin WidgetInspectorService {
   void _registerSignalServiceExtension({
     required String name,
     required FutureOr<Object?> Function() callback,
+    required RegisterServiceExtensionCallback registration,
   }) {
-    registerServiceExtension(
+    registration(
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:814]: these all need to remain calls to 
registerServiceExtension as that is prepending the name with inspector.` 
Concerning if tests are still passing with this change. I would hope at least some tests are verifying that the correct service extension names are used.

**Review by jacob314 — COMMENTED** [review]
lgtm once naming issue and testing issue are resolved.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/flutter/lib/src/widgets/widget_inspector.dart [hunk 4]
```diff
@@ -797,10 +796,12 @@ mixin WidgetInspectorService {
   void registerServiceExtension({
     required String name,
     required ServiceExtensionCallback callback,
+    required RegisterServiceExtensionCallback registerExtension,
   }) {
-    _registerServiceExtensionCallback(
+    registerServiceExtension(
```
Comment (zanderso) [inline packages/flutter/lib/src/widgets/widget_inspector.dart:801]: It looks like this infinite recursion is causing OOM failures in the presubmit checks.

**Pull request merged**